src/test/java/uk/ac/ebi/embl/template/TemplateEntryProcessorTest.java

```diff
@@ -0,0 +1,180 @@
+        templateEntryProcessor = getTemplateEntryProcessor();
+        molType = getMolTypeFromTemplateForTest(templateInfo);
+    }
+    
+
+    @Test
+    public void  testEntryProcess() throws Exception {
+
+        // Test with biosample Id
+        executeEntryProcessTest("SAMEA9403245");
+        
+        // Test with sample Id
+        executeEntryProcessTest("ERS7118926");
+        
+        // Test with sample alias "test_custom"
+        executeEntryProcessTest("test_custom");
+
+        // Test with valid organism
+        executeEntryProcessTestWithScientificName("Homo sapiens");
+
+        // Test with tacId
+        executeEntryProcessTestWithTaxId("9606");
+
+        // Test with invalid organizm
+        executeEntryProcessInvalidOrganism("JUNK");
+
+    }
+    
+    public void executeEntryProcessTest(String organismName) throws Exception{
+        
+        TemplateVariables templateVariables = getTemplateVariables(organismName);
+        Sample sample=getSampleForTest(templateEntryProcessor,templateVariables);
+        
+        TemplateProcessorResultSet templateProcessorResultSet = templateEntryProcessor.processEntry(templateInfo, molType, templateVariables,getOptions());
+
+        SourceFeature sourceFeature=templateProcessorResultSet.getEntry().getPrimarySourceFeature();
+        assertTrue(templateProcessorResultSet.getValidationResult().isValid());
+        assertEquals(sourceFeature.getTaxId(),Long.valueOf(sample.getTaxId()));
+        assertEquals(sourceFeature.getScientificName(),sample.getOrganism());
+    }
+
+    public void executeEntryProcessTestWithScientificName(String taxId) throws Exception{
+        
+        TemplateVariables templateVariables = getTemplateVariables(taxId);
+        TemplateProcessorResultSet templateProcessorResultSet = templateEntryProcessor.processEntry(templateInfo, molType, templateVariables,getOptions());
+        
+        SourceFeature sourceFeature=templateProcessorResultSet.getEntry().getPrimarySourceFeature();
+        assertTrue(templateProcessorResultSet.getValidationResult().isValid());
+        assertEquals(sourceFeature.getScientificName(),"Homo sapiens");
+    }
+
+    public void executeEntryProcessTestWithTaxId(String taxId) throws Exception{
+
+        TemplateVariables templateVariables = getTemplateVariables(taxId);
+        TemplateProcessorResultSet templateProcessorResultSet = templateEntryProcessor.processEntry(templateInfo, molType, templateVariables,getOptions());
+
+        SourceFeature sourceFeature=templateProcessorResultSet.getEntry().getPrimarySourceFeature();
+        assertTrue(templateProcessorResultSet.getValidationResult().isValid());
+        assertEquals(sourceFeature.getTaxId(),Long.valueOf(taxId));
+        assertEquals(sourceFeature.getScientificName(),"Homo sapiens");
+    }
+    
+    public void executeEntryProcessInvalidOrganism(String invalidOrganism) throws Exception{
```

Comment:
- Suggestion is to pass in the templateInfo and molType so that these test methods are template independent.
- Rename invalidOrganism to scientificName.